src/internal/connector/exchange/api/mock/mail.go

```diff
@@ -0,0 +1,67 @@
+package mock
```

Comment:
These mocks all read like they should belong to `graph/mock/service.go`.  They neither accept nor return exchange/api structures: it's all graph level.  I think it should be moved.

Reply:
`NewClient` and `NewService` are mockable version of their counterparts in `api` package under exchange. I've however moved the `CreateAdapter` to under `graph/service`.